src/Arcus.WebApi.Logging.Core/Correlation/HttpCorrelation.cs

```diff
@@ -32,21 +38,65 @@ public class HttpCorrelation : ICorrelationInfoAccessor
         /// <exception cref="ArgumentException">When the <paramref name="options"/> doesn't contain a non-<c>null</c> <see cref="IOptions{TOptions}.Value"/></exception>
         public HttpCorrelation(
-            IOptions<CorrelationInfoOptions> options,
+            IOptions<HttpCorrelationInfoOptions> options,
             IHttpContextAccessor httpContextAccessor,
             ICorrelationInfoAccessor correlationInfoAccessor,
             ILogger<HttpCorrelation> logger)
         {
             Guard.NotNull(options, nameof(options), "Requires a set of options to configure the correlation process");
             Guard.NotNull(httpContextAccessor, nameof(httpContextAccessor), "Requires a HTTP context accessor to get the current HTTP context");
             Guard.NotNull(correlationInfoAccessor, nameof(correlationInfoAccessor), "Requires a correlation info instance to set and retrieve the correlation information");
-            Guard.NotNull(logger, nameof(logger), "Requires a logger to write diagnostic messages during the correlation");
             Guard.NotNull(options.Value, nameof(options.Value), "Requires a value in the set of options to configure the correlation process");
-            
+
             _httpContextAccessor = httpContextAccessor;
             _options = options.Value;
             _correlationInfoAccessor = correlationInfoAccessor;
-            _logger = logger;
+            _logger = logger ?? NullLogger<HttpCorrelation>.Instance;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HttpCorrelation"/> class.
+        /// </summary>
+        /// <param name="options">The options controlling how the correlation should happen.</param>
+        /// <param name="correlationInfoAccessor">The instance to set and retrieve the <see cref="CorrelationInfo"/> instance.</param>
+        /// <param name="logger">The logger to trace diagnostic messages during the correlation.</param>
+        /// <param name="httpContextAccessor">The instance to have access to the current HTTP context.</param>
+        /// <exception cref="ArgumentNullException">When any of the parameters are <c>null</c>.</exception>
+        /// <exception cref="ArgumentException">When the <paramref name="options"/> doesn't contain a non-<c>null</c> <see cref="IOptions{TOptions}.Value"/></exception>
+        [Obsolete("Use the constructor overload with the " + nameof(HttpCorrelationInfoOptions) + " instead")]
+        public HttpCorrelation(
+            IOptions<CorrelationInfoOptions> options,
+            IHttpContextAccessor httpContextAccessor,
+            ICorrelationInfoAccessor correlationInfoAccessor,
+            ILogger<HttpCorrelation> logger)
+            : this(Options.Create(CreateHttpCorrelationOptions(options?.Value)), httpContextAccessor, correlationInfoAccessor, logger)
+        {
+        }
+
+        private static HttpCorrelationInfoOptions CreateHttpCorrelationOptions(CorrelationInfoOptions options)
```

Comment:
We could use AutoMapper for this, but can be in follow-up PR

Reply:
It's there only for backwards-compatible deprecated stuff. So it will not change, so I guess AutoMapper will be a bit overengineering in this case.